UI/obs-app.cpp

```diff
@@ -395,6 +396,10 @@ static void do_log(int log_level, const char *msg, va_list args, void *param)
 
 	vsnprintf(str, sizeof(str), msg, args);
 
+	string result = redact(str);
+	strncpy(str, result.c_str(), sizeof(str));
+	str[4095] = '\0';
```

Comment:
I think `strncpy_s` is available on all platforms we target now and might be a better option here.

Reply:
I think `strncpy_s` is unavailable on macOS, which instead has `strlcpy`? Not sure if there is a better option on all three platforms.

Reply:
It's part of the C11 standard and with Apple using clang and libc I'd be surprised if it wasn't there

Reply:
It's certainly undefined in Xcode on my end. From some searching, it looks like the `_s` family was an optional part of C11 and Apple does not appear to include it.

Reply:
Annex K is based on Microsoft's variant of the same functions, but nobody implements them:

* Clang hasn't
* GCC hasn't
* Microsoft hasn't ("Microsoft Visual Studio implements an early version of the APIs. However, the implementation is incomplete and conforms neither to C11")